Add Japanese Leaderboard

sitfoxfly opened the pull request:
Japanese LLMs are typically evaluated on a collection of standard benchmarks. This PR aims to organize these benchmarks into a simple reproducible framework, making it easier to consistently evaluate LLMs on the Japanese language.

Benchmarks:
- [x]  jaqket_v2
- [x] jcommonsenseqa
- [x] jnli
- [x] jsquad
- [x] marc_ja
- [x] mgsm
- [x] xlsum
- [x] xwinograd

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2439) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EleutherAI/lm-evaluation-harness?pullRequest=2439) before we can accept your contribution.<br/><hr/>**mtkachenko** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EleutherAI/lm-evaluation-harness?pullRequest=2439) it.</sub>

sitfoxfly commented:
It looks like `Tasks Modified` check will fail as the `Rakuten/JGLUE` HF dataset uses custom data loader script and, therefore, requires `trust_remote_code=True`. Could you advise on the best way to handle this issue?

baberabb commented:
Thanks for the PR. It's ok about the test, I'll update them in a seperate PR. Left a couple of comments mainly about import handling. Also, might be a good idea to create a group config like [mmmu](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/tasks/mmmu/_mmmu.yaml), if you think it's appropriate. The main difference between group and tag is that with group you get an aggregate score for all the tasks and can choose if you want to normalize it by total number of samples or number of tasks (`weight_by_size` True or False)

sitfoxfly commented on lm_eval/tasks/japanese_leaderboard/ja_leaderboard_xlsum.py:
```diff
@@ -0,0 +1,75 @@
+
+try:
+    from rouge_score import rouge_scorer, scoring
+except ImportError:
```
`ModuleNotFoundError` is raised now at the beginning of the script for the all required dependencies.

sitfoxfly commented on lm_eval/tasks/japanese_leaderboard/requirements.txt:
```diff
@@ -0,0 +1,3 @@
+emoji==2.14.0
```
Moved the dependencies to `pyproject.toml` under `lm_eval[japanese_leaderboard]`.

sitfoxfly commented:
Thanks for the review, @baberabb. I added a group config for clarity as you recommended and removed the tags as they would conflict.

I don't think it is possible to aggregate the scores across mixed metrics ("acc", "exact_match", "rouge2") with `aggregate_metric_list` at the moment? Therefore, I haven't included aggregation, though having a single score per leaderboard could be helpful.

sitfoxfly commented:
Sure thing, @baberabb, updated the task table!

Lynnzake commented:
> Sure thing, @baberabb, updated the task table!

@sitfoxfly hello, can i ask how do i eval based on the local files, I have already download the datasets from the huggingface, but i met the issue that even if i changed the `dataset_path` in yaml, the script still trying to grab datasets from huggingface.

My changes are as belows:
```bash
# cat ja_leaderboard_jaqket_v2.yaml
task: ja_leaderboard_jaqket_v2

dataset_path: /nfs-158/tujie/lm-eval/japanese_leaderboard/ja_leaderboard_jaqket_v2
dataset_name: v2.0
```
All yaml files being changed the `dataset_path`, but the script still:

```bash
2025-05-27:14:56:16,209 WARNING  [datasets.load:1600] Using the latest cached version of the module from /root/.cache/huggingface/modules/datasets_modules/datasets/Rakuten--JGLUE/982de5a6b7515b0fcf1918cd7070994fff6b4736b8455f794cc144fdcc3ed036 (last modified on Fri May 23 20:24:20 2025) since it couldn't be found locally at Rakuten/JGLUE, or remotely on the Hugging Face Hub.
Using the latest cached version of the module from /root/.cache/huggingface/modules/datasets_modules/datasets/Rakuten--JGLUE/982de5a6b7515b0fcf1918cd7070994fff6b4736b8455f794cc144fdcc3ed036 (last modified on Fri May 23 20:24:20 2025) since it couldn't be found locally at Rakuten/JGLUE, or remotely on the Hugging Face Hub.
2025-05-27:14:58:00,670 WARNING  [datasets.load:1600] Using the latest cached version of the module from /root/.cache/huggingface/modules/datasets_modules/datasets/Rakuten--JGLUE/982de5a6b7515b0fcf1918cd7070994fff6b4736b8455f794cc144fdcc3ed036 (last modified on Fri May 23 20:24:20 2025) since it couldn't be found locally at Rakuten/JGLUE, or remotely on the Hugging Face Hub.
```
How to solved this, appreciate the help!

baberabb commented:
Hi @Lynnzake! I think it should be (from [here](https://huggingface.co/docs/datasets/v1.1.3/loading_datasets.html#from-local-files))

```yaml
dataset_path: "json"
dataset_kwargs:
  data_files:
    train: ["/path/to/train/json"]
    validation: ["path/to/validation"]
```
We use `datasets.load_dataset(...)` to load the data which accepts two positional arguments (`path` and `name`) which we pass as `dataset_path` and `dataset_name`, and you can pass the kwargs through `dataset_kwargs` in the config.
